doc: Remove Visual Studio 2017 reference from readme

sipsorcery opened the pull request:
This PR was motivated by a comment in GUI PR (257) regarding a suggested improvement not being supported by VS2017.

When checking whether master can still be built with the VS2017 toolset ABI issues were encountered. Most likely due to the pre-compiled Qt binaries that are used.

It does not seem worth the effort to try and support VS2017, which would most likely require additional Qt binaries, or lengthy instructions on how to build static Qt binaries on Windows (which is very error prone and tedious).

hebasto commented:
Concept ACK on bringing the docs in line with the facts.

prayank23 commented:
I was not able to build using VS 2019 last time I tried: https://bitcoin.stackexchange.com/questions/99241/anyone-tried-building-bitcoin-core-with-visual-studio-2019

Still Concept ACK because I don't expect things to work with VS 2017 and agree with @sipsorcery that _It does not seem worth the effort to try and support VS2017_.

jarolrod commented on build_msvc/README.md:
````diff
@@ -57,19 +57,13 @@ PS >py -3 msvc-autogen.py
 msbuild /m bitcoin.sln /p:Platform=x64 /p:Configuration=Release /t:build
 ```
 
-- Alternatively open the `build_msvc/bitcoin.sln` file in Visual Studio.
````
If we don't have a note that < Visual Studio 2019 is not supported, won't this lead to people opening issues that they can't build with an earlier version?

sipsorcery replied:
I believe the leveldb appveyor file and readme are from the upstream repo. EIther way they are not used in the msvc CI for Bitcoin Core.

There is no Visual Studio 2016 application. Instead to make things super confusing Microsoft have used version 16.x for Visual Studio 2019, see https://en.wikipedia.org/wiki/Microsoft_Visual_Studio#History.

That's a good point about the build not working with earlier versions of Visual Studio. I'll adjust the PR to add a note to that effect.

jarolrod reviewed (commented):
Concept ACK

Maybe unrelated? But, what about these references to Visual Studio 2017?

https://github.com/bitcoin/bitcoin/blob/2b45cf0bcdb3d2c1de46899e30885c953b57b475/src/leveldb/.appveyor.yml#L11

https://github.com/bitcoin/bitcoin/blob/2b45cf0bcdb3d2c1de46899e30885c953b57b475/src/leveldb/README.md#L45

And here Visual Studio 2016

https://github.com/bitcoin/bitcoin/blob/2b45cf0bcdb3d2c1de46899e30885c953b57b475/build_msvc/bitcoin.sln#L3

jarolrod commented:
Approach ACK 39656f3b689c513aa3c509682e07dece85002128

I think you could squash the commits, and include that you are adding this warning in the commit message of 42cf3b3a65c20030b909c1dd6219aca7d5833299

sipsorcery commented:
I understand the preferred approach these days is to leave PR changes as individual commits. The GitHub merge feature has the option to squash commits and combine their messages.

MarcoFalke commented:
Our project doesn't use the merge button, but a specific merge script to preserve the git history and collect reviews on it. (Reviews are invalidated when the commit id changes, such as when squashing, even when done during a merge).

sipsorcery commented:
Squashed.

jarolrod commented on build_msvc/README.md:
````diff
@@ -57,19 +57,13 @@ PS >py -3 msvc-autogen.py
 ```
 
-- Alternatively open the `build_msvc/bitcoin.sln` file in Visual Studio.
+- Alternatively open the `build_msvc/bitcoin.sln` file in Visual Studio 2019.
````
while here, I guess we could fix the punctuation

```suggestion
- Alternatively, open the `build_msvc/bitcoin.sln` file in Visual Studio 2019.
```

jarolrod commented on build_msvc/README.md:
```diff
@@ -35,7 +35,7 @@ vcpkg integrate install
 Qt
 ---------------------
-In order to build the Bitcoin Core a static build of Qt is required. The runtime library version (e.g. v141, v142) and platform type (x86 or x64) must also match.
+In order to build the Bitcoin Core a static build of Qt is required. The runtime library version (e.g. v142) and platform type (x86 or x64) must also match.
```
```suggestion
In order to build Bitcoin Core, a static build of Qt is required. The runtime library version (e.g. v142) and platform type (x86 or x64) must also match.
```

jarolrod reviewed (commented):
ACK 51f17c4dbd46d25e2352e0a5cbfad4d438673a10

two minor nits, but I wouldn't want to hold this up over commas 🥃

sipsorcery committed:
Remove Visual Studio 2017 reference from readme

This PR was motivated by a comment in GUI PR (257) regarding a suggested improvement not being supported by VS2017.

When checking whether master can still be built with the VS2017 toolset ABI issues were encountered. Most likely due to the pre-compiled Qt binaries that are used.

It does not seem worth the effort to try and support VS2017, which would most likely require additional Qt binaries, or lengthy instructions on how to build static Qt binaries on Windows (which is very error prone and tedious).

Added advisory note about build not working with earlier Visual Studio versions.

Fixed grammar.

jarolrod commented:
ACK 0a331456e44ce2bdd3bfe62f7603a312905c6624

hebasto reviewed (approved):
ACK 0a331456e44ce2bdd3bfe62f7603a312905c6624, I have reviewed the code and it looks OK, I agree it can be merged.
